### ISSUE TITLE
Spam nummern online automatisch crawlen

### ISSUE THREAD

**Issue by BenutzerEinsZweiDrei** [haumacher/phoneblock#119]
Hallo bspw hatte mich diese nummer angerufen

https://www.cleverdialer.de/telefonnummer/022327018303

Das erste google ergebnis liefert direkt Gewissheit, dass es sich um SPAM handelt.

Die Nummer war aber nicht in der Blocklist.

Für einen Entwickler sollte es kein Problem sein

All diese Websites automatisch zu crawlen und 

Entsprechende nummern zur Blocklist hinzuzufügen.

Wo ist hier das Problem?

**Comment by geimist** [comment]
Warum so forsch? @haumacher erstellt das alles in seiner Freizeit und bekommt keinen Cent dafür. Ich bin sehr dankbar für seinen Einsatz, sodass ich dieses Projekt nutzen kann und denke, dass wir als User (Nutznieser) nicht in der Position sind, hier Forderungen zu stellen. Das ist ein Opensource-Projekt, wo jeder – auch du – zur Verbesserung beitragen kann.

> Die Nummer war aber nicht in der Blocklist.

Diese Nummer ist [seit dem 2. Dezember 2024, 12:10:44 UTC](https://phoneblock.net/phoneblock/nums/022327018303) in der Blacklist. Wenn du aber nur das CardDAV-Adressbuch nutzt, so kann dein Router nicht die komplette Blacklist verarbeiten. In dem Fall wäre der [Phoneblock-AB](https://phoneblock.net/phoneblock/anrufbeantworter/) eine gute Alternative.

**Comment by BenutzerEinsZweiDrei** [comment]
Sorry war nicht beabsichtigt forsch zu klingen. Entschuldigt falls der Ton unangemessen war.

**Comment by BenutzerEinsZweiDrei** [comment]
Alles klar. Also habe ich richtig verstanden, dass es fritz.box Modelle gibt, welche bei CardDAV nicht die komplette Liste verarbeiten können (oder auch zu anderen Problemen führen)? Mit neueren fritz.box Modellen funktioniert das alles vermutlich besser?

**Comment by geimist** [comment]
> Für einen Entwickler sollte es kein Problem sein All diese Websites automatisch zu crawlen und Entsprechende nummern zur Blocklist hinzuzufügen.

Guck mal [HIER](https://github.com/haumacher/phoneblock/issues/34)

**Comment by haumacher** [comment]
Das mit dem "Crawlen" ist gar nicht so einfach, weil  sich die Webseiten dagegen "wehren". PhoneBlock versucht das tatsächlich, aber die Webseiten sperren Aufrufer, die zu viele Abfragen stellen, weil sie vermuten, dass dadurch Daten "abfließen". Wenn man effektiv andere Dienste "integrieren" wollte, müsste man dieses Crawlen dezentral organisieren. Denn, die Adresse der PhoneBlock-Servers ist schon bei vielen Seiten auf der schwarzen Liste.

**Comment by geimist** [comment]
> … Wenn man effektiv andere Dienste "integrieren" wollte, müsste man dieses Crawlen dezentral organisieren. …

Das wäre eigentlich ein praktisches optionales Zusatzfeature für das Dockerimage, welches ja über all verteilt läuft. Zudem ändert sich bei vielen Usern täglich die IP.

**Comment by haumacher** [comment]
Ja, das wäre eine gute Option - man muss ja auch nur die Nummern crawlen, die bei dem Betreffenden aktiv anrufen - und sie danach auch noch an PhoneBlock reporten. PR wanted :-)

**Comment by haumacher** [comment]
Der Crawling-Code ist schon da: `de.haumacher.phoneblock.meta.MetaSearchService`

**Comment by MarkyMarkDE** [comment]
> Das mit dem "Crawlen" ist gar nicht so einfach, weil sich die Webseiten dagegen "wehren". PhoneBlock versucht das tatsächlich, aber die Webseiten sperren Aufrufer, die zu viele Abfragen stellen, weil sie vermuten, dass dadurch Daten "abfließen". Wenn man effektiv andere Dienste "integrieren" wollte, müsste man dieses Crawlen dezentral organisieren. Denn, die Adresse der PhoneBlock-Servers ist schon bei vielen Seiten auf der schwarzen Liste.

absolut korrekt, streng genommen ist das auch Contentdiebstahl, egal ob das ein legitimer Zweck ist oder nicht.
